framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2234,16 +2240,33 @@ public Void visitAnnotation(AnnotationTree tree, Void p) {
       return null;
     }
 
-    TypeElement anno = (TypeElement) TreeInfo.symbol((JCTree) tree.getAnnotationType());
+    TypeElement annoType = (TypeElement) TreeInfo.symbol((JCTree) tree.getAnnotationType());
 
-    Name annoName = anno.getQualifiedName();
+    Name annoName = annoType.getQualifiedName();
     if (annoName.contentEquals(DefaultQualifier.class.getName())
         || annoName.contentEquals(SuppressWarnings.class.getName())) {
       // Skip these two annotations, as we don't care about the arguments to them.
       return null;
     }
 
-    List<ExecutableElement> methods = ElementFilter.methodsIn(anno.getEnclosedElements());
+    if (isPreOrPostConditionAnnotation(annoName)) {
```

Comment:
Why isn't this logic in `BaseTypeVisitor#checkContractsAtMethodDeclaration`?

Reply:
I initially implemented this check in `BaseTypeVisitor#checkContractsAtMethodDeclaration` (because I also thought it was useful that the contracts are already there in the method), but that method appears to check implicit annotations, as well.